### ISSUE TITLE
[Feature Request] Add support for notifications which work without Google Play Services installed

### ISSUE THREAD

**Issue by diademiemi** [ProtonMail/proton-mail-android#96]
**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

**Comment by gahfy** [comment]
Actually, as far as I have seen, this may be quite complicated. Some solution:
* Implement a WebSocket service: This is out of Android's scope, and nothing let me think that it is scheduled
* Remove notifications: Mail application without notification, looks like nobody wants it
* Background service: That would make HTTP request every x seconds. This would ruin your battery, your data plan if it's not unlimited, the score of the app on GooglePlay, ...

The only way to have something which does not rely on Google is so totally in ProtonMail hands (or maybe in the hands of a server/websocket developer, I don't know the implementation of server side to be honest)

**Comment by diademiemi** [comment]
I can see that this is a large effort. 
In my opinion I would still think an option for a background service is a good solution. It does drain battery, but I think it could be worth it, even if it only checks for mail once every 30 minutes or something.

**Comment by diademiemi** [comment]
I think that is the perfect solution, and I hope ProtonMail adopts something like this too.

**Comment by CristianKerr** [comment]
+1

**Comment by Bluebaerchen** [comment]
yes, on the Fairmail for example you can select the intervals for checking for new mails. my choice is here 2 hours . I have 7 accounts and i can't see any battery drain. all is fine. please make ANY notifications (1 time per day at least) withot google play services. 

`(I can't understand it anyway.A privacy product, which advertises anonymous and secure communication, uses of all things GOOGLE play services and offers the app ONLY in the Play Store, where the users are tracked by google and co.)`

**Comment by onkoe** [comment]
I was immensely depressed when I booted up my shiny new LOS 18.1 installation only to find that I had to go restore my backups again; microG was purposefully left out, and yet, here I am installing it all again. I was hoping to see this fixed in the past couple of years; is there anything in particular that's holding you up?

**Comment by Bluebaerchen** [comment]
please make ANY notifications (1 time per day at least) withot google play services.

**Comment by diademiemi** [comment]
> I was immensely depressed when I booted up my shiny new LOS 18.1 installation only to find that I had to go restore my backups again; microG was purposefully left out, and yet, here I am installing it all again. I was hoping to see this fixed in the past couple of years; is there anything in particular that's holding you up?

For anyone else on LineageOS and wanting MicroG, you can install MicroG's fork of LineageOS without needing a reinstall, you can just flash over it in my experience. https://lineage.microg.org/

This at least makes notifications for ProtonMail (And other apps) work.

**Comment by onkoe** [comment]
> For anyone else on LineageOS and wanting MicroG, you can install MicroG's fork of LineageOS without needing a reinstall, you can just flash over it in my experience. https://lineage.microg.org/
> 
> This at least makes notifications for ProtonMail (And other apps) work.

Interesting, thanks for the info. Do you know if there are any "side effects" that come with these builds? I ended up using Open GApps but I'm not satisfied with my privacy.

**Comment by Bluebaerchen** [comment]
i dont have microG and i dont need microG. Thid is only 1 app, that doesn't work with notifications. I have XMPP, Telegram, Briar, Matrix, some Mail-Profiles with Fairmail. All this works fine without microG!

**Comment by onkoe** [comment]
Hi! I'm back again to report that microG + LineageOS produces notifications when Cloud Messaging is enabled. Couldn't find a good source for whether or not this worked so I tried it myself. Hope this helps someone ;)

**Comment by Bluebaerchen** [comment]
MicroG is't option for me :( proton subscription is off now and i'll just change email provider. 
agian...
(I can't understand it anyway.A privacy product, which advertises anonymous and secure communication, uses of all things GOOGLE play services and offers the app ONLY in the Play Store, where the users are tracked by google and co.)

**Comment by edwardzhxi** [comment]
need this feature, please do something, thanks
